lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -866,6 +871,80 @@ def test_source_is_empty(self):
+            'html5_sources': ['http://example.org/video.mp4'],
+        }
+        # path to subs_3_yD_cEKoCk.srt.sjson file
+        file_name = 'subs_3_yD_cEKoCk.srt.sjson'
+        # pylint: disable=no-value-for-parameter
+        test_dir = path(__file__).abspath().dirname().dirname().dirname().dirname().dirname()
+        file_path = test_dir + '/common/test/data/uploads/' + file_name
+
+        self.initialize_module(metadata=metadata)
+        item = self.store.get_item(self.item_descriptor.location)
+        with open(file_path, "r") as myfile:
+            save_to_store(myfile.read(), file_name, 'text/sjson', item.location)
+        item.sub = "3_yD_cEKoCk"
+        # subs_video.srt.sjson does not exist before calling editor_saved function
+        with self.assertRaises(NotFoundError):
+            Transcript.get_asset(item.location, 'subs_video.srt.sjson')
+        old_metadata = own_metadata(item)
+        # calling editor_saved will generate new file subs_video.srt.sjson for html5_sources
+        item.editor_saved(self.user, old_metadata, None)
+        self.assertIsInstance(Transcript.get_asset(item.location, 'subs_3_yD_cEKoCk.srt.sjson'), StaticContent)
+        self.assertIsInstance(Transcript.get_asset(item.location, 'subs_video.srt.sjson'), StaticContent)
+
+    def test_editor_saved_when_youtube_and_html5_subs_exist(self):
+        """
+        When both youtube_sub and html5_sub already exist then no new
+        sub will be generated by editor_saved function.
+        """
+        metadata = {
+            'source': 'http://youtu.be/3_yD_cEKoCk',
+            'html5_sources': ['http://example.org/video.mp4'],
+        }
+        # path to subs_3_yD_cEKoCk.srt.sjson file
+        file_name = 'subs_3_yD_cEKoCk.srt.sjson'
+        # pylint: disable=no-value-for-parameter
+        test_dir = path(__file__).abspath().dirname().dirname().dirname().dirname().dirname()
+        file_path = test_dir + '/common/test/data/uploads/' + file_name
+
+        self.initialize_module(metadata=metadata)
+        item = self.store.get_item(self.item_descriptor.location)
+        with open(file_path, "r") as myfile:
+            save_to_store(myfile.read(), file_name, 'text/sjson', item.location)
+            save_to_store(myfile.read(), 'subs_video.srt.sjson', 'text/sjson', item.location)
+        item.sub = "3_yD_cEKoCk"
+        # subs_3_yD_cEKoCk.srt.sjson and subs_video.srt.sjson already exist
+        self.assertIsInstance(Transcript.get_asset(item.location, 'subs_3_yD_cEKoCk.srt.sjson'), StaticContent)
+        self.assertIsInstance(Transcript.get_asset(item.location, 'subs_video.srt.sjson'), StaticContent)
+        old_metadata = own_metadata(item)
+        item.editor_saved(self.user, old_metadata, None)
+        # subs_3_yD_cEKoCk.srt.sjson and subs_video.srt.sjson exist after calling editor_saved function
+        self.assertIsInstance(Transcript.get_asset(item.location, 'subs_3_yD_cEKoCk.srt.sjson'), StaticContent)
```

Comment:
Can you explicitly check that `manage_video_subtitles_save` method is being called or not. Otherwise this test does't make any sense to me.